src/main/test/surfrdUtils_test/test_surfrdUtils.pf

```diff
@@ -44,21 +44,130 @@ contains
   end subroutine setup_test_collapse_routines
 
   @Test
-  subroutine test_collapse_nat_pfts()
+  subroutine test_collapse_to_dom_landunits()
+     ! Tests subroutine collapse_to_dominant when used with n_dom_landunits.
+     ! This test tries 5 values of n_dom_landunits:
+     ! 1) 0, which ends up with the same output values as were used as inputs
+     ! 2) 1, which collapses to 1 dominant landunit
+     ! 3) 2, which collapses to 2 dominant landunits
+     ! 4) 3, which collapses to 3 dominant landunits
+     ! 5) max_lunit, which behaves the same as (1) whether subroutine
+     ! collapse_to_dominant actively collapses the landunits to themselves or
+     ! simply skips the work with an if statement (current implementation).
+     !
+     ! For each value of n_dom_landunits, we test three configurations that
+     ! need to be collapsed.
+     use pftconMod, only: pftcon
+     use clm_instur, only: wt_lunit
+     use landunit_varcon, only: max_lunit, istsoil
+
+     implicit none
+     integer :: test  ! test number in loop of tests
+     integer, parameter :: begg = 2, endg = 4, tests = 5
+     integer, parameter :: n_dom_landunits(tests) = (/ 0, 1, 2, 3, max_lunit/)
+     real(r8), allocatable :: wt_expected(:,:)
+     real(r8), allocatable :: wt_in_out(:,:)  ! used in subr. call
+     real(r8) :: expctd(9)
+
+     ! Set relevant pftcon values to defaults; override where necessary
+     call pftcon%InitForTesting()
+     allocate( wt_lunit(begg:endg,istsoil:max_lunit) )
+     allocate( wt_expected(begg:endg,istsoil:max_lunit) )
+     allocate( wt_in_out(begg:endg,istsoil:max_lunit) )
+
+     ! INPUT VALUES
+     wt_lunit(begg:,:) = 0._r8  ! initialize
+     wt_lunit(begg:,1) = (/ 15._r8, 11._r8, 15._r8/)  ! landunit 1
+     wt_lunit(begg:,2) = (/  5._r8,  5._r8,  5._r8/)  ! ...
```

Comment:
landunit 3 is skipped is that on purpose?

Reply:
I noticed that the model skips landunit 3 (istcrop = 2 and then istice_mec = 4) and that's the only reason I did it. It really doesn't matter for the tests's purposes.

Reply:
Ahhh, OK that makes sense. I wonder if you should use landunit parameters to spell this out then? I suppose as you say it doesn't matter for these The Senate Judiciary committee will be held in SCR 352.  That is the Senate Committee Room 352, but would make it more obvious what landunits are being tested this way. With the hardcoded magic numbers, I have no idea.

Reply:
Ok, done.